### ISSUE TITLE
Data source for AWS Organizations account tagging

### ISSUE THREAD

**Issue by pascal-hofmann** [hashicorp/terraform-provider-aws#11511]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

#8896 added support for AWS Organizations account tagging to [`aws_organizations_account`](https://www.terraform.io/docs/providers/aws/r/organizations_account.html). 
Unfortunately there is no way to get the list of tags for an account via data source. 

* I suggest to introduce a new data source `aws_organizations_account` that exposes account details including tags. 
* Alternatively there could be a new data source `aws_organizations_account_tags`.
* Or maybe even a more generic data source `aws_resource_tags`.

Related API calls: [ListTagsForResource](https://docs.aws.amazon.com/quicksight/latest/APIReference/API_ListTagsForResource.html), [ListAccounts](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListAccounts.html)

### New or Affected Resource(s)

* new data source `aws_organizations_account` or
* new data source `aws_organizations_account_tags` or
* new data source `aws_resource_tags`

### Potential Terraform Configuration

```hcl
data "aws_organizations_account" "example" {
  filter {
    name   = "id"
    values = ["987654321234"]
  }
  filter {
    name   = "name"
    values = […]
  }
  …
}
```
```
data "aws_organizations_account_tags" "example" {
  id = "987654321234"
}
```
```
data "aws_resource_tags" "example" {
  id = "987654321234"
}
```

### References

* #8896

**Comment by AMMullan** [comment]
Can definitely see massive use-cases for this - especially if we can get a list of accounts based on a tag. Potentially we could look at expanding AWS behaviour to allow applying SCP's based on a Tag which would be amazing!

**Comment by gramsa49** [comment]
My use case for this capability is to store key information, per AWS account, that can be used to retrieve data from external systems (Terraform data source) to drive Terraform automation.  As an example, if AWS account X hosts application system Y, the AWS Orgs Account tags should have an identifier for application system Y.  This can be used to retrieve other critical information regarding cost allocations, availability goals, contact information, criticality rating, etc., which can in turn be used to drive other Terraform automation.

In our case, AWS accounts are not created using Terraform, so this information is otherwise inaccessible.

**Comment by desertmoose706** [comment]
> This functionality has been released in [v3.71.0 of the Terraform AWS Provider](https://github.com/hashicorp/terraform-provider-aws/blob/v3.71.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
> 
> For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-aws/issues/new/choose) following the template. Thank you!

I don't see this functionality in any of the releases?

**Comment by desertmoose706** [comment]
For those wondering, the data resource added is aws_organizations_resource_tags
